databroker/src/broker.rs

```diff
@@ -4233,28 +4240,23 @@ pub mod tests {
             .await
             .expect("Register datapoint should succeed");
 
-        let mut stream = broker
+        let stream = broker
             .subscribe(HashMap::from([(id1, HashSet::from([Field::Datapoint]))]))
             .await
             .expect("subscription should succeed");
 
         // Stream should yield initial notification with current values i.e. NotAvailable
-        match stream.next().await {
-            Some(next) => {
-                assert_eq!(next.updates.len(), 1);
-                assert_eq!(
-                    next.updates[0].update.path,
-                    Some("test.datapoint1".to_string())
-                );
-                assert_eq!(
-                    next.updates[0].update.datapoint.as_ref().unwrap().value,
-                    DataValue::NotAvailable
-                );
-            }
-            None => {
-                panic!("did not expect stream end")
-            }
-        }
+        let _ = stream.then(|next| async move {
+            assert_eq!(next.updates.len(), 1);
+            assert_eq!(
+                next.updates[0].update.path,
+                Some("test.datapoint1".to_string())
+            );
+            assert_eq!(
+                next.updates[0].update.datapoint.as_ref().unwrap().value,
+                DataValue::NotAvailable
+            );
+        });
```

Comment:
What's the reasoning behind the change here?

Reply:
Cant remember but it is fixed now.